File: app/src/main/java/com/database_termproject/twitter/ui/splash/SplashActivity.java
package com.database_termproject.twitter.ui.splash;

import com.database_termproject.twitter.databinding.ActivitySplashBinding;
import com.database_termproject.twitter.ui.BaseActivity;
import com.database_termproject.twitter.ui.example.ExampleActivity;
import com.database_termproject.twitter.ui.main.MainActivity;
import com.database_termproject.twitter.ui.signin.SigninActivity;

import java.util.Timer;
import java.util.TimerTask;

public class SplashActivity extends BaseActivity<ActivitySplashBinding> {
    @Override
    protected ActivitySplashBinding getBinding() {
        return ActivitySplashBinding.inflate(getLayoutInflater());
    }

    @Override
    protected void initAfterBinding() {
        // 1초 후, 이동
        new Timer().schedule(new TimerTask() {
            @Override
            public void run() {
                startNextActivity(MainActivity.class);
                finish();
            }
        }, 1000);
    }
}

File: app/src/main/java/com/database_termproject/twitter/ui/splash/SplashActivity.kt
package com.database_termproject.twitter.ui.splash

import android.os.Handler
import com.database_termproject.twitter.databinding.ActivitySplashBinding
import com.database_termproject.twitter.ui.BaseActivity
import com.database_termproject.twitter.ui.main.ExampleActivity
import com.database_termproject.twitter.ui.main.MainActivity


class SplashActivity : BaseActivity<ActivitySplashBinding>(ActivitySplashBinding::inflate) {
    override fun initAfterBinding() {
        // 1초 후, Main Activity로 이동
        Handler().postDelayed(Runnable {
            startNextActivity(ExampleActivity::class.java)
            finish()
        }, 1000)
    }
}